### PR TITLE
RL-based Maze Solver PR

### PR DESCRIPTION
**Pull request by tushdon2**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,97 @@
-# RL-based-Maze-Solver
\ No newline at end of file
+# RL-based Maze Solver
+Open Projects 2021
+
+![Input Maze Solution](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/Images%20and%20Videos/Images/Input%20maze1%20and%20solution.png)
+<br>fig1 <i>Input Maze Solution</i>
```
Comment (theobscuredev) [inline README.md]: Center align all the images using the `<p>` tag and align attribute. For eg:
`<p align="center">`
  `LED Matrix`
  `<i>LED Matrix</i>`
`</p>`

File: README.md [hunk 2]
```diff
@@ -1 +1,97 @@
...
+Our project inspires us to solve many problems, like Autonomous Mobile Robot Obstacle Prevention using Q-learning.</p>
+
+![Image Processing](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/Images%20and%20Videos/Images/Input%20maze2_image%20processing.png)
+<br>fig3 <i>Image Processing</i>
+
+<!-- ## Mechanical Aspect of the Design -->
+
+<!-- ## Electronics Aspect of the Design -->
+
+## Software Aspect of the Design
```
Comment (theobscuredev) [inline README.md:27]: Justify the text for remaining content like Software Aspects, Cost Structure, Limitations, Future Improvements etc which is currently missing.

**Review by theobscuredev — CHANGES_REQUESTED** [review]
@tushdon2  Hey, nice work. Please do the suggested changes.

File: README.md (hunk 1 above)
Comment (tushdon2) [reply README.md]: centred all the images

File: README.md (hunk 2 above)
Comment (tushdon2) [reply README.md:27]: justified the paragraphs

Comment (tushdon2) [reply README.md:27]: updated the README

**Comment by theobscuredev** [comment]
@tushdon2 Lemme know once you've made the final changes. Make sure to tag me (@) in this thread.

**Comment by tushdon2** [comment]
@theobscuredev
We have made all the required changes.

**Comment by theobscuredev** [comment]
Nice work!
@SanjeevKrishnan I suggest for merge.

**Review by SanjeevKrishnan — CHANGES_REQUESTED** [review]
Few Changes are required
1. In README.md, under Software Aspect of the Design, add a sub section "Working" and explain the need and working of algorithms used in detail.
2. File names of Images and Videos need to be more generalised. For example in Images and Videos/Videos/Main Program.mp4 doesn't make any sense. I suggest Main Demonstration.mp4 or simply Demonstration.mp4
@theobscuredev Kindly follow this up

**Comment by tushdon2** [comment]
@SanjeevKrishnan @theobscuredev 
The required changes have been made.
Hope we met your expectations. Kindly look into the repo and do the needful.

**Review by SanjeevKrishnan — CHANGES_REQUESTED** [review]
Few changes are still required!
1. Rename "README ImgPreprocess.md" as "README.md"
2. Rename "Code_4.0.ipynb" as "Maze Solving Code.ipynb"
3. In README.md, do changes as suggested below

File: src/README ImgPreprocess.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    
+    for i in range(self.h):
+        if self.nim[i, self.w -1] == 0: 
+            l.append(i)
+            l.append(self.w -1)
+            count += 1
+            if count == 2 : return l
+```
+
+#### 9. This [ImgPreprocess.py](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/src/ImgPreprocess.py) file is then imported by the [Maze Solving Code(Code_4.0.ipynb)](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/src/Code_4.0.ipynb).
```
Comment (SanjeevKrishnan) [inline src/README ImgPreprocess.md]: Do necessary changes here, after renaming the files as suggested in the previous comment.

Comment (tushdon2) [reply src/README ImgPreprocess.md]: @SanjeevKrishnan  
We have made all the changes you asked for.
Please review.

**Comment by tushdon2** [comment]
@theobscuredev 
@SanjeevKrishnan

**Review by SanjeevKrishnan — APPROVED** [review]
Great Work team!!
PR looks good. Merging the PR

**Pull request merged**